app/src/components/inputs/inputDropdown/inputDropdown.jsx

```diff
@@ -30,74 +30,97 @@ const cx = classNames.bind(styles);
     options: PropTypes.array,
     multiple: PropTypes.bool,
     selectAll: PropTypes.bool,
     disabled: PropTypes.bool,
-    active: PropTypes.bool,
     onChange: PropTypes.func,
     onFocus: PropTypes.func,
     onBlur: PropTypes.func,
   };
 
   static defaultProps = {
     displayedValue: '',
+    value: '',
     options: [],
     multiple: false,
     selectAll: false,
     disabled: false,
-    active: false,
     onChange: () => {},
     onFocus: () => {},
     onBlur: () => {},
   };
+  state = {
+    opened: false,
+  };
   componentDidMount() {
     document.addEventListener('click', this.handleClickOutside);
   }
   componentWillUnmount() {
     document.removeEventListener('click', this.handleClickOutside);
   }
   onClickSelectBlock = (e) => {
-    e.stopPropagation();
-    this.props.active
-      ? (() => this.props.multiple && this.props.onBlur())()
-      : this.props.onFocus();
+    if (!this.props.disabled) {
+      this.setState({ opened: !this.state.opened });
+      e.stopPropagation();
+      this.state.opened
+        ? this.props.onBlur()
+        : this.props.onFocus();
+    }
   };
   handleClickOutside = (e) => {
     if (this.node.contains(e.target) && this.props.multiple) {
       return;
     }
     this.props.onBlur();
   };
+
+  handleChange = (selectValue) => {
+    const { multiple, value, onChange } = this.props;
+    if (multiple) {
+      if (value.indexOf(selectValue) > -1) {
+        onChange(value.filter(item => item !== selectValue));
+      } else {
+        onChange([...value, selectValue]);
+      }
+    } else {
+      onChange(selectValue);
+    }
+    this.setState({ opened: !this.state.opened });
+  };
+  handleAllClick = () => {
+    if (this.props.value.length !== this.props.options.length) {
+      this.props.onChange(this.props.options.map(item => item.value));
+    } else {
+      this.props.onChange([]);
+    }
+  };
+
   renderOptions() {
     return this.props.options.map(option => (
       <DropdownOption
-        key={option.id}
-        id={option.id}
+        key={option.value}
+        value={option.value}
         disabled={option.disabled}
-        active={option.active}
-        text={option.text}
+        selected={this.props.value.indexOf(option.value) > -1}
+        label={option.label}
         multiple={this.props.multiple}
+        onChange={option.disabled ? () => {} : this.handleChange}
       />
     ));
   }
 
   render() {
-    const classes = cx({
-      dropdown: true,
-      opened: this.props.active,
-    });
     return (
-      <div ref={(node) => { this.node = node; }} className={classes}>
+      <div ref={(node) => { this.node = node; }} className={cx('dropdown', { opened: this.state.opened })}>
         <div className={cx({ 'select-block': true, disabled: this.props.disabled })} onClick={this.onClickSelectBlock}>
           <span className={cx('value')}>{ this.props.displayedValue }</span>
```

Comment:
I think it's better to calculate value to display from the `value` prop instead of defining a special prop for that